pgmpy/estimators/TreeSearch.py

```diff
@@ -100,87 +96,134 @@ def estimate(self):
-                D.add_edge(class_node, node)
+        n_jobs: int (default: -1)
+            Number of jobs to run in parallel. `-1` means use all processors.
 
-            return DAG(D)
+        edge_weights: str or function (default: mutual info)
+            Method to use for computing edge weights. Options are:
+                1. 'mutual_info': Mutual Information Score.
+                2. 'adjusted_mutual_info': Adjusted Mutual Information Score.
+                3. 'normalized_mutual_info': Normalized Mutual Information Score.
+                4. function(array[n_samples,], array[n_samples,]): Custom function.
 
-    def chow_liu(self, data, root_node, n_jobs):
-        # create a fully connected graph over the independent variables
-        def calculate_mi(u_index, v_index):
-            return (
-                u_index,
-                v_index,
-                mutual_info_score(data.iloc[:, u_index], data.iloc[:, v_index]),
+        show_progress: boolean
+            If True, shows a progress bar for the running algorithm.
+
+        Returns
+        -------
+        model: `pgmpy.base.DAG` instance
+            The estimated model structure.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import pandas as pd
+        >>> from pgmpy.estimators import TreeSearch
+        >>> values = pd.DataFrame(np.random.randint(low=0, high=2, size=(1000, 5)),
+        ...                       columns=['A', 'B', 'C', 'D', 'E'])
+        >>> est = TreeSearch(values, root_node='B')
+        >>> model = est.estimate(estimator_type='chow-liu')
+        """
+        # Step 0: Check for edge weight computation method
+        if edge_weights == "mutual_info":
+            edge_weights = mutual_info_score
+        elif not isinstance(edge_weights, callable):
+            raise ValueError(
+                "edge_weights either needs to be 'mutual_info' or a function of form fun(array, array)"
             )
 
-        total_cols = len(data.columns)
-        pbar = tqdm(
-            combinations(enumerate(data.columns), 2),
-            total=(total_cols * (total_cols - 1) / 2),
-        )
-        pbar.set_description("Building tree")
+        # Step 1: Compute edge weights for a fully connected graph.
+        n_vars = len(data.columns)
+        if show_progress:
+            pbar = tqdm(
+                combinations(data.columns, 2), total=(n_vars * (n_vars - 1) / 2)
+            )
+            pbar.set_description("Building tree")
+        else:
+            pbar = combinations(data.columns, 2)
 
-        vals = Parallel(n_jobs=n_jobs)(
-            delayed(calculate_mi)(u_index=u_index, v_index=v_index)
-            for (u_index, u), (v_index, v) in pbar
+        vals = Parallel(n_jobs=n_jobs, prefer="threads")(
+            delayed(edge_weights)(data.loc[:, u], data.loc[:, v]) for u, v in pbar
         )
+        weights = np.zeros((n_vars, n_vars))
+        weights[np.triu_indices(n_vars, k=1)] = vals
```

Comment:
More of a question than review.  About `Parallel`, does it preserve ordering? It must be, otherwise, the indices in these 2 blocks of code would not match up.

```
pbar = combinations(data.columns, 2)
vals = Parallel(n_jobs=n_jobs, prefer="threads")(
   delayed(edge_weights)(data.loc[:, u], data.loc[:, v]) for u, v in pbar
)
```

`weights[np.triu_indices(n_vars, k=1)] = vals`

Reply:
Yes, it does preserve the order.